zerver/models.py

```diff
@@ -3193,6 +3193,14 @@ def where_starred() -> str:
         # will generate a query involving `flags & 2 = 2`, which doesn't match our index.
         return "flags & 2 <> 0"
 
+    @staticmethod
+    def where_flag_is_present(flagattr: Bit) -> str:
+        return f"flags & {1 << flagattr.number} <> 0"
+
+    @staticmethod
+    def where_flag_is_absent(flagattr: Bit) -> str:
+        return f"flags & {1 << flagattr.number} = 0"
+
     @staticmethod
     def where_active_push_notification() -> str:
```

Comment:
Can we rewrite `where_active_push_notification` and `where_starred` / `where_unread` in terms of these functions, as a prep commit?

Reply:
Done.